crates/daphne/src/vdaf/mod.rs

```diff
@@ -228,6 +229,28 @@ impl AsMut<[u8]> for VdafVerifyKey {
     }
 }
 
+fn upgrade_codec_error(error: CodecErrorDraft09) -> CodecError {
```

Comment:
There is already a `from_codec_error` on line 104 which implements exactly this function. These should be merged, and I wonder if `upgrade_codec_error` and `downgrade_codec_error` could be macros?

Reply:
I don't know how I could make them with macros

Reply:
So I just had a go, and you can get fairly close, but I couldn't get all the way there. 
Something like
```rust
macro_rules! make_from_codec_error {
    ($e:ident, $f: ident, $t: ident, $($variants:expr),+) => {
        fn $e(input: $f) -> $t {
            match input {
                $(
                    $f::$variants => $t::$variants,
                )*
            }
        }
    }
}
```

Reply:
yeah but you end up having to write more code than if done manually, no? Either way I deleted `downgrade_*` because I realized encoding using the prio_d9 wasn't worth it anyway